File: problem_set_2/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

// output Grade 16+
// output Before Grade 1
//  output Grade X

// prototype
float cal_letters(string text);
float cal_words(string text);
float cal_sentence(string text);
int cal_index(float letters, float words, float sentences);

int main(void)
{
    string text = get_string("Text: ");
    float letters = cal_letters(text);
    float words = cal_words(text);
    float sentences = cal_sentence(text);
    // printf("%s\n", text);
    // printf("%f\n",letters);
    // printf("%f\n",words);
    // printf("%f\n",sentences);
    int index = cal_index(letters, words, sentences);
    // printf("%i\n",index);
    if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (index <= 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

// count letters
float cal_letters(string text)
{
    float letters = 0;
    for (int i = 0, length = strlen(text); i < length; i++)
    {
        if (isalpha(text[i]))
        {
            letters += 1;
        }
    }

    return letters;
    // L is average number of letters per 100 words
    // S is the average number of sentences per word
    // index = 0.0588 * L - 0.296 * S - 15.8
}

// count words
float cal_words(string text)
{
    float words = 1;
    for (int i = 0, length = strlen(text); i < length; i++)
    {
        if (isblank(text[i]))
        {
            words += 1;
        }
    }
    return words;
}

// count sentences
float cal_sentence(string text)
{
    float sentence = 0;
    for (int i = 0, length = strlen(text); i < length; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentence += 1;
        }
    }
    return sentence;
}

// count index
int cal_index(float letters, float words, float sentences)
{
    float L = (letters / words) * 100;
    float S = (sentences / words) * 100;
    float index = (0.0588 * L) - (0.296 * S) - 15.8;
    // printf("%f\n", L);
    // printf("%f\n", S);
    // printf("%f\n", index);
    return round(index);
}

File: problem_set_6/sentimental-readability/readability.py
import cs50


def main():
    text = cs50.get_string("Text :")
    letters = cal_letters(text)
    words = cal_words(text)
    sentences = cal_sentences(text)
    index = cal_index(letters, words, sentences)

    if index >= 16:
        print("Grade 16+")
    elif index <= 1:
        print("Before Grade 1")
    else:
        print(f"Grade {index}")


def cal_letters(text):
    letters = 0
    for i in range(len(text)):
        if text[i].isalpha():
            letters += 1
    return letters


def cal_words(text):
    words = 1
    for i in range(len(text)):
        if text[i].isspace():
            words += 1
    return words


def cal_sentences(text):
    sentence = 0
    for i in range(len(text)):
        if text[i] in [".", "!", "?"]:
            sentence += 1

    return sentence


def cal_index(letters, words, sentences):
    L = (letters / words) * 100
    S = (sentences / words) * 100
    index = (0.0588 * L) - (0.296 * S) - 15.8

    return round(index)


if __name__ == "__main__":
    main()
